lib/ansible/modules/identity/keycloak/keycloak_client.py

```diff
@@ -722,13 +722,20 @@ def main():
     changeset = dict()
 
     for client_param in client_params:
-        # lists in the Keycloak API are sorted
         new_param_value = module.params.get(client_param)
+
+        # some lists in the Keycloak API are sorted, some are not.
         if isinstance(new_param_value, list):
-            try:
-                new_param_value = sorted(new_param_value)
-            except TypeError:
-                pass
+            if client_param in ['attributes']:
+                try:
+                    new_param_value = sorted(new_param_value)
+                except TypeError:
+                    pass
+        # Unfortunately, the ansible argument spec checker introduces variables with null values when
+        # they are not specified
+        if client_param == 'protocol_mappers':
+            new_param_value = map(lambda x: dict((k, v) for (k, v) in x.items() if x[k] is not None), new_param_value)
```

Comment:
Oh, sorry, I missed this was using map. I think this will throw an AttributeError on Python 3 when you use camel(). You can just use `dict((k, v) for k, v in new_param_value.items() if new_param_value[k] is not None)`, right?

Reply:
I changed the code to use a list comprehension of dict comprehensions to avoid the use of map. :)